10up/classifai
# Initial integration into the new Command Palette

### Description of the Change

Gutenberg v15.6.0 shipped with a new Command Palette that can be triggered within the site editor (by using the keyboard shortcut command+k). This feature should be shipping with WP 6.3. This PR does some initial integrations into this Command Palette but there are certainly more things we could add here (or in follow-up PRs).

At the moment, there are 4 commands we add here:

1. A link to the general ClassifAI settings page
2. Ability to trigger excerpt generation
3. Ability to trigger title generation
4. A link that takes you to the new stand-alone image generation screen

https://github.com/10up/classifai/assets/916738/0302f108-7305-468c-af95-5dbf25d516df

### How to test the Change

1. Ensure you're running the Gutenberg plugin, v15.6.0 or greater
2. Check this PR out and run `npm install && npm run build`
3. Create a new post or edit an existing post
5. Click into the post content (I found I couldn't trigger the command center until I clicked on something) and then use the shortcut command+k
6. In the modal that pops up, search for `ClassifAI: Generate excerpt`, `ClassifAI: Generate title`, `ClassifAI: Generate image`, or `ClassifAI settings`
7. Click on all three separately and ensure they work

### Changelog Entry

> Added - Initial integration with the new Command Palette API

### Credits

Props @dkotter

### Checklist:
